### PR TITLE
chore: [IAI-23] Remove obsolete fetch polyfill

### PR DESCRIPTION
**Pull request by pietro909**
## Short description
For this PR to be safe, I decided first to add extensive tests around our `fetch()` wrappers _and then_ proceed to remove the obsolete polyfill.
This led to a major refactoring.
I've also upgraded Node to `14.17.4` (current LTS) while working on it.

This PR also closes https://www.pivotaltracker.com/n/projects/2048617/stories/170819445

## List of changes proposed in this pull request
- Replace [mock-http-server](https://www.npmjs.com/package/mock-http-server) with [mockttp](https://www.npmjs.com/package/mockttp) which offers a similar interface but comes with **types included**
- Move initialization code for Jest to `jestSetup.js`
- review the entire test suite in `ts/utils/__tests__/fetch.test.ts` testing only the public functions
- Remove obsolete [whatwg-fetch](https://www.npmjs.com/package/whatwg-fetch)
- Add a handy helper for running individual tests with low noise [in package.json](https://github.com/pagopa/io-app/compare/IAI-23-remove-obsolete-fetch-polyfills?expand=1#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R36)
- Upgrade `abortcontroller-polyfill` to `1.7.3` 

## How to test
Besides the automated testing, I've installed and run the app using login, messages, services, profile, to look for possible exceptions. No errors noted so far.

Please **do test edge cases** you may be aware of.

Tested on:
* iPhone 12 (14.5) - emulator
* Redmi Note 8 Pro

### CODE REVIEW

File: ts/utils/fetch.ts [hunk 1]
```diff
@@ -90,24 +78,16 @@ export function retryLogicForTransientResponseError(
...
  * This is a fetch with timeouts, constant backoff and with the logic
- * that handles 404s as transient errors, this "fetch" must be passed to
- * createFetchRequestForApi when creating "getPaymentId"
+ * that handles 404s as transient errors.
  */
 export const constantPollingFetch = (
   shouldAbort: Promise<boolean>,
   retries: number,
   delay: number,
   timeout: Millisecond = fetchTimeout
```
Comment (pietro909) [inline ts/utils/fetch.ts:87]: After a brief chat with @fabriziofff it turns out that the timeout parameter applies only _per request_. Will update accordingly.

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ release_tag: &release_tag /^[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+)?$/
 defaults_js: &defaults_js
   <<: *defaults
   docker:
-    - image: cimg/node:12.22
+    - image: cimg/node:14.17.4
```
[the remark this answers is not in the document]
Comment (pietro909) [reply .circleci/config.yml:24]: Good catch!
I've checked it now and it appears that we don't really need a Node image since the scripts are using only Bash and Python 🤔 
Do you know why is using such an image?

[the remark this answers is not in the document]
Comment (pietro909) [reply .circleci/config.yml:24]: https://github.com/pagopa/io-app/pull/3402/commits/b70219edd011a9e265d08814570559855c16c5b8

Trying with `python:3.8` like the other python job.

[the remark this answers is not in the document]
Comment (fabriziofff) [reply .circleci/config.yml:24]: Yes, that was my fault, definitely better!

**Review by fabriziofff — APPROVED** [review]
